tests/func/test_ls.py

```diff
@@ -139,8 +139,8 @@ def test_ls_repo_with_path_dir_dvc_only_empty(tmp_dir, dvc, scm):
     tmp_dir.dvc_gen(DVC_STRUCTURE, commit="dvc")
     tmp_dir.scm_gen({"folder/.keep": "content"}, commit="add .keep")
 
-    with pytest.raises(PathMissingError):
-        Repo.ls(os.fspath(tmp_dir), path="folder", dvc_only=True)
+    files = Repo.ls(os.fspath(tmp_dir), path="folder", dvc_only=True)
```

Comment:
Discussed with @karajan1001 that this fixes https://github.com/iterative/dvc/issues/6106 but #5841 is actually not a bug, but an intended behaviour as `folder` in this test doesn't exist from DvcFileSystem. So a proper solution here is to only use DvcFileSystem in case of `dvc_only=True` instead of using `RepoFileSystem` like we do right now.

Reply:
@efiop Doesn't with `--dvc-only` flag should this exception been thrown out? in #5841 no `--dvc-only` flag on it.